File: Search/python/03_베스트셀러.py
# fast campus 강의
# https://www.acmicpc.net/problem/1302
# 1

N = int(input())

dic = {}
for _ in range(N):
    s = input()
    if s not in dic:
        dic[s] = 1
    else:
        dic[s] += 1

max_value = max(dic.values())
result = []

for key, value in dic.items():
    if value == max_value:
        result.append(key)

print(sorted(result)[0])

File: Search/c++/03_베스트셀러.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/1302
// 1

#include <iostream>
#include <map>

using namespace std;

int N;
map<string, int> book;

int main() 
{
    cin >> N;
    for (int i = 0; i < N; i++) {
        string s;
        cin >> s;
        book[s] += 1;
    }

    int max_value = 0;
    string result;
    for (auto b : book) {
        if (max_value < b.second) {
            max_value = b.second;
            result = b.first;
        }
    }

    cout << result;
}